Wikipedia/Code/EditLinkViewController.swift

```diff
@@ -0,0 +1,204 @@
+import UIKit
+
+protocol EditLinkViewControllerDelegate: AnyObject {
+    func editLinkViewController(_ editLinkViewController: EditLinkViewController, didTapCloseButton button: UIBarButtonItem)
+    func editLinkViewController(_ editLinkViewController: EditLinkViewController, didFinishEditingLink displayText: String?, linkTarget: String)
+    func editLinkViewController(_ editLinkViewController: EditLinkViewController, didFailToExtractArticleTitleFromArticleURL articleURL: URL)
+    func editLinkViewControllerDidRemoveLink(_ editLinkViewController: EditLinkViewController)
+}
+
+class EditLinkViewController: ViewController {
+    weak var delegate: EditLinkViewControllerDelegate?
+
+    typealias Link = SectionEditorWebViewMessagingController.Link
+    private let link: Link
+    private let siteURL: URL
+    private var articleURL: URL
+
+    private let articleCell = ArticleRightAlignedImageCollectionViewCell()
+    private let dataStore: MWKDataStore
+
+    private var navigationBarVisibleHeightObservation: NSKeyValueObservation?
+
+    @IBOutlet private weak var contentView: UIView!
+    @IBOutlet private weak var scrollViewTopConstraint: NSLayoutConstraint!
+    @IBOutlet private weak var displayTextLabel: UILabel!
+    @IBOutlet private weak var displayTextView: UITextView!
+    @IBOutlet private weak var displayTextViewHeightConstraint: NSLayoutConstraint!
+    @IBOutlet private weak var linkTargetLabel: UILabel!
+    @IBOutlet private weak var linkTargetContainerView: UIView!
+    @IBOutlet private weak var linkTargetContainerViewHeightConstraint: NSLayoutConstraint!
+    @IBOutlet private weak var activityIndicatorView: UIActivityIndicatorView!
+    @IBOutlet private weak var removeLinkButton: AutoLayoutSafeMultiLineButton!
+    @IBOutlet private var separatorViews: [UIView] = []
```

Comment:
Thought these had to be IBOutletCollections 🤔 seems to work fine without though.

Reply:
Looks like `@IBOutletCollection` is not supported in Swift? 🤔 Trying to declare `@IBOutletCollection` produces and error and dragging from the xib, `Outlet Collection` shows up as `@IBOutlet`

Reply:
🤦‍♀️ ha no problem - a difference that I'm just now noticing 5 years later.